WeakAuras/Profiling.lua

```diff
@@ -360,3 +366,160 @@ function WeakAuras.PrintProfile()
   end
   popup:Show()
 end
+
+RealTimeProfilingWindow = CreateFrame("Frame", nil, UIParent)
+RealTimeProfilingWindow.width = 400
+RealTimeProfilingWindow.height = 300
+RealTimeProfilingWindow.barHeight = 20
+RealTimeProfilingWindow.titleHeight = 15
+RealTimeProfilingWindow.bars = {}
+RealTimeProfilingWindow:SetMovable(true)
+
+local texture = "Interface\\DialogFrame\\UI-DialogBox-Background"
+function RealTimeProfilingWindow:AddBar(num, name, map, total)
+  local bar = self.bars[name]
+
+  -- hide bar if it's going to be out of background window
+  if self.titleHeight + self.barHeight * num > self.height then
+    if bar then
+      bar:Hide()
+    end
+    return
+  end
+
+  local pct = 100 * map.elapsed / total
+
+  if not bar then
+    bar = CreateFrame("FRAME", nil, self)
+    self.bars[name] = bar
+    Mixin(bar, SmoothStatusBarMixin)
+
+    local fg = bar:CreateTexture(nil, "ARTWORK")
+    fg:SetSnapToPixelGrid(false)
+    fg:SetTexelSnappingBias(0)
+    fg:SetTexture(texture)
+    fg:SetDrawLayer("ARTWORK", 0)
+    fg:Show()
+    bar.fg = fg
+
+    local bg = bar:CreateTexture(nil, "ARTWORK")
+    bg:SetSnapToPixelGrid(false)
+    bg:SetTexelSnappingBias(0)
+    bg:SetTexture(texture)
+    bg:SetDrawLayer("ARTWORK", -1)
+    bg:SetAllPoints()
+    bg:Show()
+    bar.bg = bg
+
+    bar:SetSize(self.width, self.barHeight)
+    if not bar.txtName then
+      local txtName = bar:CreateFontString(nil, "OVERLAY", "GameFontNormal")
+      bar.txtName = txtName
+      txtName:SetPoint("TOPLEFT", bar, "TOPLEFT")
+      txtName:SetPoint("BOTTOMRIGHT", bar, "BOTTOMRIGHT", -30)
+      txtName:SetJustifyH("LEFT")
+    end
+    if not bar.txtPct then
+      local txtPct = bar:CreateFontString(nil, "OVERLAY", "GameFontNormal")
+      bar.txtPct = txtPct
+      txtPct:SetPoint("TOPLEFT", bar, "TOPRIGHT", -50, 0)
+      txtPct:SetPoint("BOTTOMRIGHT", bar, "BOTTOMRIGHT")
+      txtPct:SetJustifyH("RIGHT")
+    end
+    bar:Show()
+  end
+  bar:ClearAllPoints()
+  bar:SetPoint("TOPLEFT", self, "TOPLEFT", 0, - (num - 1) * self.barHeight - self.titleHeight)
+  bar.fg:ClearAllPoints()
+  bar.fg:SetPoint("TOPLEFT", bar)
+  bar.fg:SetSize(self.width / 100 * pct, self.barHeight)
```

Comment:
Consider mixing in the Smoothed progress mixin from blizzard and using smoothed progress. 

It’s optional but I think it could look quite nice.